### PR TITLE
net, refactor: extract Network and BIP155Network logic to node/network

### PR DESCRIPTION
**Pull request by jonatack**
This extracts the `Network` and `BIP155Network` logic to `node/network`.  The code has been living between `netaddress` and `netbase` and some compilation units include these large files when they only need a `Network` enum or related method.  Separating the code to a standalone unit in `node` improves code separation and helps with using only what is needed.

I verified the `include` headers with https://cirrus-ci.com/task/6749578737745920 generated by https://github.com/bitcoin/bitcoin/pull/27385/commits/8f647a65d3484c7acd2d97f4b055c582d7734b6f while this was in draft and carefully narrowed them down to the most relevant ones.

Possible todos for a follow-up: upgrade `Network` to an `enum class`, e.g. `NET_I2P` becomes `Network::I2P` and https://github.com/bitcoin/bitcoin/pull/27385/commits/5cfa3fb8b5815aaf96483a63526e5f0bf3c0a06b.

### CODE REVIEW

File: src/test/fuzz/addrman.cpp [hunk 1]
```diff
@@ -26,6 +28,14 @@
 namespace {
 const BasicTestingSetup* g_setup;
 
+static const std::map<uint8_t, uint8_t> NET_LEN_MAP = {
```
Comment (fanquake) [inline src/test/fuzz/addrman.cpp:31]: In 9bcbbc68682c1a92316d6a2dd8142839c58fe247 (fuzz, refactor: hoist net_len_map in addrman fuzzer to constant): Isn't this already a constant? Seems to just be moving the code out of the function where it's used?

File: src/netbase.cpp [hunk 2]
```diff
@@ -377,7 +377,7 @@ bool Socks5(const std::string& strDest, uint16_t port, const ProxyCredentials* a
     vSocks5.push_back(SOCKSVersion::SOCKS5); // VER protocol version
     vSocks5.push_back(SOCKS5Command::CONNECT); // CMD CONNECT
```
Comment (ccdle12) [inline src/netbase.cpp:378]: Could be a candidate for another PR - since we are changing `SOCKS5Atyp` to an enum  class, would it make sense to also change the following enums to enum classes `SOCKSVersion, SOCKS5Command, SOCKS5Method` and `SOCKS5Reply`?

**Comment by ccdle12** [comment]
tACK for creating smaller compilation units. 

I'm just curious, for the motivation of future changes, should there be more node networking related logic moved to the node sub folder (e.g. logic related to the node making decisions on networking but not the actual networking implementations themselves) or is it simply classes/enums that are called frequently in isolation, in different parts of the code base but they exist in bigger files with unused imports like in `netbase` and `netaddress`?

**Comment by ccdle12** [comment]
> Rebased 3rd_place_medal
> 
> @ccdle12 Thanks! Mind re-acking? As to motivation, I think it's a mix of both reasons you mentioned, with the first one being the direction to go when doing so -- see `doc/design/libraries.md` and merged changes like [c741d74](https://github.com/bitcoin/bitcoin/commit/c741d748d4d9836940b99091cc7be09c65efcb79).

Of course! re-tACK at 8efd76b, thanks for explaining the motivation and pointing out the design/doc :)

**Comment by fanquake** [comment]
> and helps with using only what is needed, which may reduce build size

What do you mean by "build size"?
